averbis/core/_rest_client.py

```diff
@@ -139,6 +149,66 @@ def successful(self):
         return self.exception is None
 
 
+class ResourceContainer:
+
+    def __init__(
+            self,
+            client: "Client",
+            name: str,
+            scope: str,
+            base_url: str,
+    ):
+        self.client = client
+        self.name = name
+        self.scope = scope
+        self.base_url = base_url
+
+    @experimental_api
+    def list_resources(self) -> List[str]:
+        """
+        HIGHLY EXPERIMENTAL API - may soon change or disappear.
+        List paths of resources in the container.
+        """
+        # noinspection PyProtectedMember
+        return self.client._list_container_resources(self.name, self.base_url)
+
+    @experimental_api
+    def export_resources(self, target_path: Union[str, Path]) -> None:
+        """
+        HIGHLY EXPERIMENTAL API - may soon change or disappear.
+        Export the whole resource container as a zip file.
+        """
+        # noinspection PyProtectedMember
+        self.client._export_container_resources(target_path, self.name, self.base_url)
+
+    @experimental_api
+    def export_resource(self, target_path: Union[str, Path], resource_path: str) -> None:
+        """
+        HIGHLY EXPERIMENTAL API - may soon change or disappear.
+        Export a specific resource file from within the container at the given resources_path.
+        """
+        # noinspection PyProtectedMember
+        self.client._export_resource(target_path, self.name, self.base_url, resource_path)
+
+    @experimental_api
+    def upsert_resource(self, resource_file_path: Union[str, Path], resource_path: str) -> None:
```

Comment:
What about using more distinct names here, e.g. use `upload_file_path` instead of `resource_file_path`?